packages/app/src/app/overmind/namespaces/editor/internalActions.ts

```diff
@@ -83,6 +83,10 @@ export const saveCode: AsyncAction<{
     return;
   }
 
+  if (state.editor.currentId === 'new' && !state.editor.isForked) {
```

Comment:
I think this will only work in the react one :/

Reply:
Can you explain why you check for the currentID?

Reply:
Thanks for the quick review.

> I think this will only work in the react one :/

Good catch, I forgot to test for other templates

> Can you explain why you check for the currentID?

Consider doing `CMD + S` on an already saved app, `isForked` will be false and would just do `return` and doesn't call the API.


Is there a way to know when the app would be forked? We can add that logic over here and that would work for other templates as well.

Reply:
Would we be able to exit the flow if an error occurs in the fork itself? Something like an early return? That would seem like the most probably solution to me.

Reply:
`isForked = false` tells that there was an error while forking and we can use it to return early. And this would work currently if the fork is unsuccessful the only issue that exists we need to fix the condition for `currentId === 'new'`.

Reply:
I got an idea do let me know if this sounds good: 

We can create a state variable called `hasForkError` and set it to `true`  when the Fork API throws or returns an error. And exit the flow when this is `true` something like this: 

```
if (hasForkError) {
   return;
}
```

cc @SaraVieira  @CompuIves

Reply:
Hmm, would it be possible to do this:

```
try {
    await actions.editor.internal.forkSandbox({
      sandboxId: state.editor.currentId,
    });
} catch (e) {
  return;
}
```

here:

https://github.com/CompuIves/codesandbox-client/blob/7e67db3c72b7c2099d33296c437c4a60be47d8ac/packages/app/src/app/overmind/factories.ts#L74-L76

?

Reply:
Ah, I see we'd need to do that at multiple places then, here as well:

https://github.com/CompuIves/codesandbox-client/blob/7e67db3c72b7c2099d33296c437c4a60be47d8ac/packages/app/src/app/overmind/namespaces/editor/internalActions.ts#L301

Reply:
Great @CompuIves  this works properly. Thanks for helping me with this. I missed the `factories` file, now I know how things work when we save. You can check the implementation if it looks good.